optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,456 @@
+        for param_name in search_space.keys():
+            param_distribution = search_space[param_name]
+            parents_param = [p.params[param_name] for p in parents]
+
+            # categorical data operates on uniform crossover
+            if isinstance(search_space[param_name], CategoricalDistribution):
+                param = _swap(parents_param[0], parents_param[-1], rng.rand(), swapping_prob)
+                child[param_name] = param
+                continue
+
+            trans = _SearchSpaceTransform({param_name: param_distribution})
+            transes.append(trans)
+            distributions.append(param_distribution)
+            param_names.append(param_name)
+            for parent_index, trial in enumerate(parents):
+                param = trans.transform({param_name: trial.params[param_name]})[0]
+                parents_not_categorical_params[parent_index].append(param)
+
+        xs = np.array(parents_not_categorical_params)
+
+        if crossover_name == "uniform":
+            params_array = _uniform(xs[0], xs[1], rng, swapping_prob)
+        elif crossover_name == "blxalpha":
+            alpha = 0.5
+            params_array = _blxalpha(xs[0], xs[1], rng, alpha)
+        elif crossover_name == "sbx":
+            if len(study.directions) == 1:
+                eta = 2
+            else:
+                eta = 20
+            params_array = _sbx(xs[0], xs[1], rng, distributions, eta)
+        elif crossover_name == "vsbx":
+            if len(study.directions) == 1:
+                eta = 2
+            else:
+                eta = 20
+            params_array = _vsbx(xs[0], xs[1], rng, eta)
+        elif crossover_name == "undx":
+            sigma_xi = 0.5
+            sigma_eta = 0.35 / np.sqrt(len(xs[0]))
+            params_array = _undx(xs[0], xs[1], xs[2], rng, sigma_xi, sigma_eta)
+        elif crossover_name == "undxm":
+            _m = len(xs) - 2
+            _n = len(xs[0])
+            assert _n - _m > 0
+
+            sigma_xi = 1 / np.sqrt(_m)
+            sigma_eta = (
+                0.35
+                * np.sqrt(_m + 1)
+                * np.sqrt(3)
+                / np.sqrt(_n - _m)
+                / np.sqrt(_m + 2)
+                / np.sqrt(2)
+            )
+            params_array = _undxm(xs, rng, sigma_xi, sigma_eta)
+        elif crossover_name == "spx":
+            epsilon = np.sqrt(len(xs[0]) + 2)
+            params_array = _spx(xs, rng, epsilon)
+        else:
+            assert False
+
+        _params = [
+            trans.untransform(np.array([param])) for trans, param in zip(transes, params_array)
+        ]
+        for param in _params:
+            for param_name in param.keys():
+                child[param_name] = param[param_name]
+
+        if _is_contained(child, search_space):
+            break
+
+    return child
+
+
+def _selection(
+    crossover_name: str,
+    study: Study,
+    parent_population: Sequence[FrozenTrial],
```

Comment:
The name of this argument differs from that of `_select_parent` function. Is it intended? I think it is better for code readability to use the same name.